apps/common/lib/lexical/ast.ex

```diff
@@ -436,34 +474,15 @@ defmodule Lexical.Ast do
   """
   @spec expand_aliases(
           alias_segments() | module(),
-          Document.t(),
+          Analysis.t() | Document.t(),
           Position.t() | {Position.line(), Position.character()}
         ) ::
           {:ok, module()} | :error
-  def expand_aliases(module_or_segments, %Document{} = document, %Position{} = position) do
-    with {:ok, quoted} <- fragment(document, position) do
-      expand_aliases(module_or_segments, document, quoted, position)
-    end
-  end
-
-  def expand_aliases(module_or_segments, %Document{} = document, {line, column}) do
-    expand_aliases(module_or_segments, document, Position.new(document, line, column))
-  end
-
-  @spec expand_aliases(alias_segments() | module(), Document.t(), Macro.t(), Position.t()) ::
-          {:ok, module()} | :error
-  def expand_aliases(module, %Document{} = document, quoted_document, %Position{} = position)
-      when is_atom(module) and not is_nil(module) do
-    module
-    |> Module.split()
-    |> Enum.map(&String.to_atom/1)
-    |> expand_aliases(document, quoted_document, position)
-  end
-
-  def expand_aliases(segments, %Document{} = document, quoted_document, %Position{} = position)
+  def expand_aliases(segments, analysis_or_document, %Position{} = position)
```

Comment:
this function still has a pretty significant performance footgun if you pass in a document. I think we should be putting the AST (and possibly the analysis) into the document store, and retrieve it from there. This will, I think, require moving AST into `lexical_shared`